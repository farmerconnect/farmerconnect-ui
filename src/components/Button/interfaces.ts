import { ButtonHTMLAttributes } from 'react';
import CustomStyles from '../CustomStyles';

export interface IButton {
  colors: IButtonColors;
}

export interface IButtonColors {
  primary: string;
  secondary: string;
  tertiary?: string;
}

export interface IButtonProps extends CustomStyles, ButtonHTMLAttributes<HTMLButtonElement> {
  backgroundColor?: string;
  color?: string;
  hoverColor?: string;
}
