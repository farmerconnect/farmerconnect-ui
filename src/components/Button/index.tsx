import React from 'react';
import { IButtonProps } from './interfaces';
import Container from './styles';

const Button: React.FC<IButtonProps> = ({ backgroundColor, color, customStyles, children, ...props }) => (
  <Container backgroundColor={backgroundColor} color={color} customStyles={customStyles} {...props}>
    {children}
  </Container>
);

export default Button;
