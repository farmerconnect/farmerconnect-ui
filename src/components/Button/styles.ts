import styled from 'styled-components';
import { IButton, IButtonProps } from './interfaces';

export const buttonDefaultStyles: IButton = {
  colors: {
    primary: '#00E394',
    secondary: '#FFFFFF',
    tertiary: '#00AF72',
  },
};

const Container = styled.button<IButtonProps>`
  background-color: ${(props) =>
    props.backgroundColor || props.theme?.button?.colors?.primary || buttonDefaultStyles.colors.primary};
  border: 0px;
  border-radius: 12px;
  color: ${(props) => props.color || props.theme?.button?.colors?.secondary || buttonDefaultStyles.colors.secondary};
  cursor: pointer;
  font-weight: bold;
  min-height: 2rem;
  min-width: 7rem;
  outline: none;
  padding: 13px 46px;
  text-align: center;
  font-family: 'Red Hat Text', sans-serif;
  font-size: 0.875rem;

  &:hover {
    background-color: ${(props) =>
      props.hoverColor || props.theme?.button?.colors?.tertiary || buttonDefaultStyles.colors.tertiary};
  }

  ${(props) => props.customStyles}
`;

export default Container;
