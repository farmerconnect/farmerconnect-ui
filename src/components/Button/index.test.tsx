import React from 'react';
import { shallow } from 'enzyme';
import Button from './';

describe('Button component', () => {
  test('it matches the snapshot', () => {
    const container = shallow(<Button>Test</Button>);
    expect(container).toMatchSnapshot();
  });
});
