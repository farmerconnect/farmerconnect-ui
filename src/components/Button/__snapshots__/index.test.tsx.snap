// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component it matches the snapshot 1`] = `
<styled.button>
  Test
</styled.button>
`;
